### PR TITLE
chore: add Dependabot for dependency updates

### PR DESCRIPTION
**Pull request by nschonni**
## Motivation

As discussed in https://github.com/facebook/docusaurus/issues/3552
Maybe you want to go with Renovate Bot in the longer term, but at least this would get something going.
Kept updates to weekly to reduce some of the noise, but maybe you want to increase it to monthly and just use the manual trigger on the Insights tab to trigger manually

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

N/A

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

**Comment by Josh-Cena** [comment]
This doesn't need configuration from the admin, yeah? I haven't used Dependabot around the time it was integrated into GH.

Also, we can go even less frequently than that. I would investigate the actual configuration.

**Comment by nschonni** [comment]
Monthly is the max right now https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/configuration-options-for-dependency-updates#scheduleinterval
I'd probably let the GitHub Actions one stay more frequent, since most of the actions are just pointing to the major versions anyway

**Comment by nschonni** [comment]
I don't believe any admin setup is required with the config file now, but it was in the past when it was done through the separate app

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+updates:
+  - package-ecosystem: github-actions
+    directory: '/'
+    schedule:
+      interval: weekly
+    open-pull-requests-limit: 99
+  - package-ecosystem: npm
+    directory: '/'
+    schedule:
+      interval: weekly
```
Comment (nschonni) [inline .github/dependabot.yml:11]: If you want it less frequently
```suggestion
      interval: monthly
```

**Review by Josh-Cena — COMMENTED** [review]
Overall I don't like Dependabot and I use Renovate for my projects, because Dependabot has no batch-updates and ends up sneding 3 PRs to upgrade one monorepo (e.g. ts-eslint) and creates huge noise in our history. It also usually means waiting for Dependabot to resolve conflicts after merging every PR. But... seems we haven't progressed much with integrating Renovate.

I would leave the decision to @slorber

File: .github/dependabot.yml (hunk 1 above)
Comment (Josh-Cena) [reply .github/dependabot.yml:11]: Yes, let's go for monthly 👍

**Comment by Josh-Cena** [comment]
@nschonni Is there a way for us to ignore certain lockfiles and only update the one at the root? I noticed that dependabot opens four PRs at a time because we have three example packages that aren't supposed to be modified

**Comment by Josh-Cena** [comment]
Oh, I see. That sounds good

**Comment by slorber** [comment]
Until we set up Renovate, that seems reasonable to use this bot for now as I don't need to ask Joel to set up anything.

Will we get flooded with a lot of immediate PRs just after the merge?

**Comment by Josh-Cena** [comment]
> Will we get flooded with a lot of immediate PRs just after the merge?

We probably will, considering the size of our yarn.lock... Let's do a manual relock first, shall we?

**Comment by slorber** [comment]
Yes, I'm doing one right now

**Comment by Josh-Cena** [comment]
Did one myself as well, noticed a few type incompatibilities, probably the `@types/` files were updated

**Comment by slorber** [comment]
👍  let's give it a try

**Comment by Josh-Cena** [comment]
Oh, forgot to configure the Dependabot PR labels :D

**Comment by Josh-Cena** [comment]
No, actually, we don't want Dependabot to add labels: that's too much noise in our changelog. The default ones are fine

**Pull request merged**